ddtrace/contrib/internal/openai/utils.py

```diff
@@ -248,16 +273,27 @@ def _is_async_generator(resp):
 
 
 def _loop_handler(span, chunk, streamed_chunks):
-    """Sets the openai model tag and appends the chunk to the correct index in the streamed_chunks list.
-
-    When handling a streamed chat/completion response, this function is called for each chunk in the streamed response.
     """
-    if span.get_tag("openai.response.model") is None:
-        span.set_tag("openai.response.model", chunk.model)
-    for choice in chunk.choices:
-        streamed_chunks[choice.index].append(choice)
-    if getattr(chunk, "usage", None):
-        streamed_chunks[0].insert(0, chunk)
+    Sets the openai model tag and appends the chunk to the correct index in the streamed_chunks list.
+    When handling a streamed chat/completion and responses response,
+    this function is called for each chunk in the streamed response.
+    """
+    # Handle both completion and response types
+    if hasattr(chunk, "type") and chunk.type.startswith("response."):
+        if span.get_tag("openai.response.model") is None:
+            span.set_tag("openai.response.model", chunk.response.model)
+        if hasattr(chunk, "response"):
+            streamed_chunks[0].append(chunk.response)
+        if getattr(chunk, "usage", None):
+            streamed_chunks[0].insert(0, chunk.response)
+    else:
```

Comment:
```suggestion
            streamed_chunks[0].insert(0, chunk.response)
            return
```
We only care about one or the other case, so we can just return early and unindent the following blocks